### ISSUE TITLE
LRP

### ISSUE THREAD

**Issue by albermax** [albermax/innvestigate#2]
* Restructure code.
* Add pooling layers. (max pooling is covered with the gradient, sum pooling should work with LRP rules; means need to add reverse hook).
* Add merge layers; i.e. add reverse hooks.
* (still needed?) Add LNR layer to innvestigate.layers and create a LRP rule to handle it.
* Add check for all keras layer if compatible, if not raise exception.
* Get to work for all application networks.
* Organize module: which rules to keep, divide DT and LRP?

**Comment by JaeDukSeo** [comment]
thank you for the contributions!

**Comment by cae67** [comment]
Hi all! Have these issues with getting NaNs in neural networks containing batch normalization layers been resolved?

I am currently getting NaN values for some samples with lrp.alpha1beta0 and am trying to figure out why.

**Comment by adrhill** [comment]
Hi @cae67, the issue persists. Subscribing to issue #292 will notify you when we have resolved it.
